### PR TITLE
Update community docs

### PR DESCRIPTION
**Pull request by cuijulian**
Continued from https://github.com/kiegroup/optaweb-employee-rostering/pull/237
Rebased on `master` and updated for the refactor.

### CODE REVIEW

**Comment by Christopher-Chianelli** [comment]
Will merge when green

File: employee-rostering-docs/src/main/asciidoc/ProjectStructure/ProjectStructure-section.adoc [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+* *Shift* is the _planning entity_, where is the defined the relationship with the _planning variable_ employee. Other important fields: `spot`, `rotationEmployee`, `startDateTime`, `endDateTime`.
+* *Employee* is the _planning variable_, it's identified by the `name` and has a set of skills (`skillProficiencySet`).
+* *Roster* is the _planning solution_, `employeeList` is the list of employees (the range of values that can be assigned to the _planning variable_), the field `score` holds the score (3 levels: hard, medium, soft), the other problem facts are: `skillList`, `spotList`, `employeeAvailabilityList`, `rosterParametrization`, `rosterState`, `shiftList`.
+
+== Constraints
+
+The constraints are defined in the *employee-rostering-backend* module, with the implementation of the backend REST
+service.
+
+*Worth noting:*
```
Comment (ge0ffrey) [inline employee-rostering-docs/src/main/asciidoc/ProjectStructure/ProjectStructure-section.adoc]: Remove this line. It's verbosity

File: employee-rostering-docs/src/main/asciidoc/ProjectStructure/ProjectStructure-section.adoc [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+
+TIP: The first line of the `when` clause is a technique to dynamically change the weight of the constraint. If `rotationEmployeeMatchWeight` is 0 the constraint is disregarded.
+
+IMPORTANT: In general, employees desire to work following a regular schedule: a rotation plan. This represents a
+starting point for the actual schedule that is influenced by other factors (e.g. temporary unavailability). For this
+reason, all Shifts are initialized with a `rotationEmployee`.
+
+Condition: There a shift that is assigned to an employee which is not the rotation employee.
+
+Action: The soft score is decreased by _rotationEmployeeMatchWeight_ units.
```
Comment (Christopher-Chianelli) [inline employee-rostering-docs/src/main/asciidoc/ProjectStructure/ProjectStructure-section.adoc]: We also need explanations for contracts, and GwtJavaTimeWorkaroundUtil do not exist any more

File: employee-rostering-docs/src/main/asciidoc/ProjectStructure/ProjectStructure-section.adoc [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+[source,ruby]
+----
+rule "Required skill for a shift"
+    when
+        Shift(
+                employee != null,
+                !getEmployee().hasSkills(getSpot().getRequiredSkillSet()))
+    then
+        scoreHolder.addHardConstraintMatch(kcontext, -100);
+end
```
Comment (ge0ffrey) [inline employee-rostering-docs/src/main/asciidoc/ProjectStructure/ProjectStructure-section.adoc]: Do we want to include all DRL implementations in the documentation?
This increases the maintenance burden: every constraint change must now be duplicated into the docs.

I don't think the docs reader needs to see the DRL. An explanation of the constraint should be enough. Wdyt?

Comment (Christopher-Chianelli) [reply employee-rostering-docs/src/main/asciidoc/ProjectStructure/ProjectStructure-section.adoc]: True, but docs on certain DRL rules might make the reader more comfortable in learning how to modify the example to include their rule set.

**Comment by Christopher-Chianelli** [comment]
Wait for green

**Pull request merged**